pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -466,21 +474,19 @@ def allocate_budget(
         else:
             budgets_size = self.budgets_to_optimize.sum().item()
 
-        # 5. Create an initial guess
-        initial_guess = np.ones(budgets_size) * (total_budget / budgets_size)
-        initial_guess = initial_guess.astype(self._budgets_flat.type.dtype)
-
-        if minimize_kwargs is None:
-            minimize_kwargs = self.DEFAULT_MINIMIZE_KWARGS.copy()
-        else:
-            # Merge with defaults (preferring user-supplied keys)
-            minimize_kwargs = {**self.DEFAULT_MINIMIZE_KWARGS, **minimize_kwargs}
+        # 5. Construct the initial guess (x0) if not provided
+        if "x0" not in minimize_kwargs:
```

Comment:
I don't think x0 should be hidden in minimize kwargs, I would make it an explicit argument of `allocate_budget`